src/sql/src/catalog.rs

```diff
@@ -275,6 +275,14 @@ pub trait SessionCatalog: fmt::Debug + ExprHumanizer + Send + Sync + ConnectionR
     /// Returns the set of supported AWS PrivateLink availability zone ids.
     fn aws_privatelink_availability_zones(&self) -> Option<BTreeSet<String>>;
 
+    /// Returns the Materialize role for assuming customer's role
+    /// for AWS Connection.
+    fn aws_external_connection_role(&self) -> Option<String>;
+
+    /// Returns the External ID prefix to be used when using AWS connection
+    /// to connect to customer's AWS account.
+    fn aws_external_id_prefix(&self) -> Option<String>;
```

Comment:
I think you should be able to avoid this by not storing the AWS info in the connection config; see future comment.

Reply:
done